### PR TITLE
Added mentions and emojis to ckeditor

### PR DESCRIPTION
**Pull request by poradz**
image
image
image

### CODE REVIEW

File: public_html/layouts/basic/modules/Vtiger/resources/BasicSearch.js [hunk 1]
```diff
@@ -65,20 +65,23 @@ jQuery.Class('Vtiger_BasicSearch_Js', {}, {
...
 		params.mode = 'showSearchResults';
 		params.limit = this.reduceNumberResults;
 		params.html = this.returnHtml;
 		if (app.getParentModuleName()) {
-			params.parent = app.getParentModuleName()
+			params.parent = app.getParentModuleName();
 		}
-		if (this.mainConatiner.find('input[data-operator]').length && this.mainConatiner.find('input[data-operator]').data('operator') != '') {
-			params.operator = this.mainConatiner.find('input[data-operator]').data('operator');
+		params.operator = 'FulltextBegin';
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/BasicSearch.js]: ?

File: public_html/layouts/basic/modules/Settings/LayoutEditor/resources/LayoutEditor.js [hunk 2]
```diff
@@ -1821,11 +1821,13 @@ $.Class('Settings_LayoutEditor_Js', {}, {
 			}).done(function (data) {
 				app.showModalWindow(data, function (modalContainer) {
 					const customConfig = {
-						toolbar: 'Min',
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Settings/LayoutEditor/resources/LayoutEditor.js]: ?

File: public_html/layouts/resources/Fields.js [hunk 3]
```diff
@@ -378,30 +380,113 @@ App.Fields = {
...
+			}
+
+			/**
+			 * Get mention data (invoked by ck editor mentions plugin)
+			 * @param {object} opts
+			 * @param {function} callback
+			 * @param {string} searchModule
+			 */
+			getMentionData(opts, callback, searchModule = '-') {
+				let basicSearch = new Vtiger_BasicSearch_Js();
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/resources/Fields.js]: ?

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 4]
```diff
@@ -243,6 +243,7 @@ jQuery.Class("Vtiger_List_Js", {
 						return;
 					}
 					app.showModalWindow(data, function (data) {
+						app.event.trigger("MassEditModal.AfterLoad", data, massActionUrl);
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]:
```suggestion
						app.event.trigger('MassEditModal.AfterLoad', data, massActionUrl);
```

File: modules/Vtiger/views/BasicAjax.php [hunk 5]
```diff
@@ -123,6 +123,24 @@ public function showSearchResults(\App\Request $request)
...
+			$searchKey = $request->get('value');
+			$queryGenerator = new \App\QueryGenerator($searchModule);
+			$queryGenerator->setFields(['id']);
+			$queryGenerator->addNativeCondition(['or',
+				['like', 'user_name', '%' . $searchKey . '%', false],
+				['like', 'last_name', '%' . $searchKey . '%', false],
+				['like', 'first_name', '%' . $searchKey . '%', false],
+			]);
+			$query = $queryGenerator->createQuery();
+			$rows = $query->limit(100)->all();
```
Comment (rskrzypczak) [inline modules/Vtiger/views/BasicAjax.php]: Add to config

Comment (rskrzypczak) [inline modules/Vtiger/views/BasicAjax.php]: dataReader

File: public_html/layouts/basic/modules/Vtiger/resources/List.js [hunk 6]
```diff
@@ -1988,6 +1988,13 @@ jQuery.Class("Vtiger_List_Js", {
 			}
 		});
 	},
+	registerMassActionModalEvents() {
+		app.event.on('MassEditModal.AfterLoad', (data, container) => {
+			if (container.is('#add-comment__container') || container.is('#sendSmsContainer')) {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/List.js]: ?

**Pull request merged**